### PR TITLE
fix(SCT-509): safely handle residents with partial addresses

### PR DESCRIPTION
**Pull request by chrishutchinson**
**What**  
Some resident's have partial addresses (e.g. missing postcode). Our UI doesn't gracefully handle these situations, until now!

**Why**  
[SCT-509](https://hackney.atlassian.net/browse/SCT-509)

### CODE REVIEW

File: components/Search/results/ResidentsTable.tsx [hunk 1]
```diff
@@ -35,9 +35,13 @@ const ResultEntry = (person: LegacyResident): React.ReactElement => {
         {dateOfBirth && new Date(dateOfBirth).toLocaleDateString('en-GB')}
       </td>
       <td className="govuk-table__cell">
-        {address?.address && truncate(address.address, 4)}
+        {address?.address && truncate(address.address || '', 4)}
```
Comment (Renny4Real) [inline components/Search/results/ResidentsTable.tsx:38]: Is `truncate` similar to a substring method and we want it to cut it down to four letters?

**Review by Renny4Real — COMMENTED** [review]
Looks good 🎉 

Would it be possible to have screenshots of before/after UI changes?

I might have been looking in the wrong place (list of results when you search for a person) but I couldn't see any visual differences when I swapped between main and this branch.

**Comment by chrishutchinson** [comment]
> Looks good 🎉
> 
> Would it be possible to have screenshots of before/after UI changes?
> 
> I might have been looking in the wrong place (list of results when you search for a person) but I couldn't see any visual differences when I swapped between main and this branch.

@Renny4Real good question – no visual changes here, just that the app now doesn't throw an error when this data is rendered (which was causing unintended log outs).

File: components/Search/results/ResidentsTable.tsx (hunk 1 above)
Comment (chrishutchinson) [reply components/Search/results/ResidentsTable.tsx:38]: Looks like it cuts down to a certain number of words, in this case 4. I haven't actually changed this, just made it work with a null `address.address` value

**Review by Renny4Real — APPROVED** [review]
👍🏾

**Pull request merged**